src/ppx_blob.ml

```diff
@@ -1,15 +1,12 @@
-open Migrate_parsetree
-open Ast_408
-
-let str ?loc ?attrs s = Ast_helper.Exp.constant ?loc ?attrs (Pconst_string (s, None))
-
-let location_errorf ~loc =
-  Format.ksprintf (fun err ->
-    raise (Location.Error (Location.error ~loc err))
-  )
+open Ppxlib
 
 let find_file_path ~loc file_name =
-  let dirname = Location.absolute_path loc.Location.loc_start.pos_fname |> Filename.dirname in
+  let dirname =
+    let loc_fname = loc.Location.loc_start.pos_fname in
+    if Filename.is_relative loc_fname then
+      Filename.dirname (Filename.concat (Sys.getcwd ()) loc_fname)
+    else Filename.dirname loc_fname
+  in
```

Comment:
@jeremiedimino, is there a better replacement for `Location.absolute_path` that is exposed by `Ppxlib`?

Reply:
I haven't looked in details, but `Location.absolute_path` might be a good candidate for addition in ppxlib. Could you open a ticket on github.com/ocaml-ppx/ppxlib?

Reply:
Thanks, done.

Reply:
```suggestion
  let dirname = Ocaml_common.Location.absolute_path loc.Ocaml_common.Location.loc_start.pos_fname |> Filename.dirname
```
Would using `Ocaml_common` temporarily solve the issue?